src/control/tools/SplineHandler.cpp

```diff
@@ -77,123 +94,113 @@ auto SplineHandler::onKeyEvent(GdkEventKey* event) -> bool {
         return false;
     }
 
-    int pointCount = stroke->getPointCount();
-    double zoom = xournal->getZoom();
-    double radius = radiusConst / zoom;
-    double x = stroke->getX() - radius;
-    double y = stroke->getY() - radius;
-    double w = stroke->getElementWidth() + 2 * radius;
-    double h = stroke->getElementHeight() + 2 * radius;
+    Rectangle rect = this->computeRepaintRectangle();
 
     switch (event->keyval) {
         case GDK_KEY_Escape: {
-            if (pointCount > 0) {
-                // remove dynamically changing point at cursor position
-                stroke->deletePoint(pointCount - 1);
-                this->redrawable->rerenderRect(x, y, w, h);
-            }
-            finalizeSpline();
+            this->redrawable->repaintRect(rect.x, rect.y, rect.width, rect.height);
+            this->finalizeSpline();
             return true;
         }
         case GDK_KEY_BackSpace: {
-            if (pointCount > 1) {
-                // delete last non dynammically changing point
-                stroke->deletePoint(pointCount - 2);
+            if (this->getKnotCount() > 0) {
+                this->deleteLastKnotWithTangent();
             }
             break;
         }
-        case GDK_KEY_Up: {
-            movePoint(0, -1);
+        case GDK_KEY_Up: {  // Shift + Up if page should not be moved outside of presentation mode
+            this->movePoint(0, -1);
             break;
         }
-        case GDK_KEY_Down: {
-            movePoint(0, 1);
+        case GDK_KEY_Down: {  // Shift + Down if page should not be moved outside of presentation mode
+            this->movePoint(0, 1);
             break;
         }
         case GDK_KEY_Right: {
-            movePoint(1, 0);
+            this->movePoint(1, 0);
             break;
         }
         case GDK_KEY_Left: {
-            movePoint(-1, 0);
+            this->movePoint(-1, 0);
+            break;
+        }
+        case GDK_KEY_r:
+        case GDK_KEY_R: {  // r like "rotate"
+            double angle = (event->state == GDK_SHIFT_MASK) ? -5 : 5;
```

Comment:
The event state is a bit field, so you need to use
```suggestion
            double angle = (event->state & GDK_SHIFT_MASK) ? -5 : 5;
```